coral-trino/src/test/java/com/linkedin/coral/trino/rel2trino/HiveToTrinoConverterTest.java

```diff
@@ -994,4 +994,19 @@ public void testSqlSelectAliasAppenderTransformerWithoutTableAliasPrefix() {
         + "WHERE \"tablea\".\"a\" > 5";
     assertEquals(expandedSql, expected);
   }
+
+  @Test
+  public void testIntCastToBigIntDuringComparison() {
+    // We're testing that a comparison between INT and BIGINT sees a cast on the more restrictive type to the
+    // less restrictive type and not the other way around. In other words, the INT is cast to BIGINT.
+    RelNode relNode = TestUtils.getHiveToRelConverter().convertSql(
+        "SELECT CASE WHEN a_integer = a_bigint THEN 'abc' ELSE 'def' END FROM test.table_from_utc_timestamp");
```

Comment:
Are we using the best table for this test case `table_from_utc_timestamp`? Table name might be misleading since it indicates it has to do with timestamp?

Reply:
Added a new test table `tableInt` in rel2trino test utils.